### PR TITLE
[RFC] Extend PE Fieldset - Stage 2 Proposal

### PR DESCRIPTION
**Pull request by peasead**
- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)? Yes
- Have you followed the [contributor guidelines](https://github.com/elastic/ecs/blob/master/CONTRIBUTING.md)? Yes
- For proposing substantial changes or additions to the schema, have you reviewed the [RFC process](https://github.com/elastic/ecs/blob/master/rfcs/README.md)? Yes
- If submitting code/script changes, have you verified all tests pass locally using `make test`? Yes
- If submitting schema/fields updates, have you generated new artifacts by running `make` and committed those changes? Yes
- Is your pull request against master? Unless there is a good reason otherwise, we prefer pull requests against master and will backport as needed. Yes
- Have you added an entry to the [CHANGELOG.next.md](https://github.com/elastic/ecs/blob/master/CHANGELOG.next.md)? Yes

### CODE REVIEW

**Comment by peasead** [comment]
Pinging to see if there are any issues to be addressed.

File: rfcs/text/0014-extend-file-pe.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # 0014: Extend the PE field set
 
-- Stage: **1 (draft)**
+- Stage: **2 (draft)**
```
Comment (ebeahan) [inline rfcs/text/0014-extend-file-pe.md:3]:
```suggestion
- Stage: **2 (candidate)**
```

File: rfcs/text/0014-extend-file-pe.md [hunk 2]
```diff
@@ -77,17 +78,217 @@ Stage 1: Provide a high-level description of example sources of data. This does
 
 <!--
 Stage 2: Included a real world example source document. Ideally this example comes from the source(s) identified in stage 1. If not, it should replace them. The goal here is to validate the utility of these field changes in the context of a real world example. Format with the source name as a ### header and the example document in a GitHub code block with json formatting.
+https[://]c4a08b8258a848efb518dcea18a1d11e[.]us-central1[.]gcp[.]cloud[.]es[.]io/goto/51d883b52d4ad99d9200c1b25cf1b5c9
```
Comment (ebeahan) [inline rfcs/text/0014-extend-file-pe.md:81]: Is this something we want to add to the comments here, or is it just something that was left in unintentionally?

File: rfcs/text/0014-extend-file-pe.md [hunk 3]
```diff
@@ -77,17 +78,217 @@ Stage 1: Provide a high-level description of example sources of data. This does
...
 -->
-
+```
+"file": {
+  "pe": {
+    "machine_type": 332,
+    "flattened": {
```
Comment (ebeahan) [inline rfcs/text/0014-extend-file-pe.md:87]: Based on the definitions [here](https://github.com/elastic/ecs/blob/master/rfcs/text/0014/pe.yml), both `debug` and `resources` would be directly underneath the `pe.*` top-level?

File: rfcs/text/0014-extend-file-pe.md [hunk 4]
```diff
@@ -77,17 +78,217 @@ Stage 1: Provide a high-level description of example sources of data. This does
...
+    "file_version": "7.4.0.8215",
+    "product": "2345王牌输入法",
+    "imports": [
+      {
+        "name": "GetStdHandle",
+        "library_name": "KERNEL32.dll",
+        "type": "function"
+      }
+    ],
+    "comments": "pinyin.2345.cc",
```
Comment (ebeahan) [inline rfcs/text/0014-extend-file-pe.md:123]: I don't see `comments` in the proposed extended `pe.*` fields or the current ones. Is this a field to add?

File: rfcs/text/0014-extend-file-pe.md [hunk 5]
```diff
@@ -77,17 +78,217 @@ Stage 1: Provide a high-level description of example sources of data. This does
...
+    "overlay": {
```
Comment (ebeahan) [inline rfcs/text/0014-extend-file-pe.md:124]: Same observation as `comments`, `overlay` isn't a field I've seen included before.

File: rfcs/text/0014-extend-file-pe.md [hunk 6]
```diff
@@ -77,17 +78,217 @@ Stage 1: Provide a high-level description of example sources of data. This does
...
+      "chi2": 26219.931640625,
+      "filetype": "Data",
+      "entropy": 7.256883144378662,
+      "offset": 480768,
+      "size": 14512,
+      "md5": "9a620f3583cc366f44e66df2a0f949a3"
+    },
+    "description": "2345王牌输入法-特殊符号输入器",
+    "creation_date": "2021-01-26T09:13:07.000Z",
+    "compilers": [
```
Comment (ebeahan) [inline rfcs/text/0014-extend-file-pe.md:134]: Again looking at the fields [defined so far](https://github.com/elastic/ecs/blob/master/rfcs/text/0014/pe.yml), `compiler.name` and `compiler.version` are defined (not `compilers`). Right now neither is indicated to be a list.

Looks like we need to revisit?

File: rfcs/text/0014-extend-file-pe.md [hunk 7]
```diff
@@ -77,17 +78,217 @@ Stage 1: Provide a high-level description of example sources of data. This does
...
+    ],
+    "sections": [
+      {
+        "chi2": 1829004.63,
+        "virtual_address": 4096,
+        "entropy": 6.49,
+        "flags": "rx",
+        "name": ".text",
+        "raw_size": 260096,
+        "virtual_size": 259926,
```
Comment (ebeahan) [inline rfcs/text/0014-extend-file-pe.md:160]: Neither `sections.virtual_size` nor `sections.md5` are defined under `sections` currently.

File: rfcs/text/0014-extend-file-pe.md [hunk 8]
```diff
@@ -77,17 +78,217 @@ Stage 1: Provide a high-level description of example sources of data. This does
...
+        "md5": "c001ed496e2cad5bb280d0d0bd97d4b4"
+      }
+    ],
+    "compile_timestamp": "2021-01-26T09:13:07.000Z",
+    "imphash": "12b9b9807742673c75c934de534f2014",
+    "original_file_name": "2345PinyinSymbol.exe",
+    "company": "版权所有(c) 2020, 2345移动科技",
+    "main_icon": {
```
Comment (ebeahan) [inline rfcs/text/0014-extend-file-pe.md:168]: `main_icon` also not currently defined.

File: rfcs/text/0014-extend-file-pe.md [hunk 9]
```diff
@@ -77,17 +78,217 @@ Stage 1: Provide a high-level description of example sources of data. This does
...
+      "hash": {
+        "dhash": "e0c8f0b0b0acc071",
+        "md5": "8f852053f4dc483d5e25b2a663a4b0b4"
+      }
+    },
+    "entry_point": 165321,
+    "hash": {
+      "authentihash": "f489ddcf530ae777496441f1409989015ab9e32e57f5f5d7e51e6908ee1dc503"
```
Comment (ebeahan) [inline rfcs/text/0014-extend-file-pe.md:176]: `authentihash` right now is defined at `pe.authentihash` and not `pe.hash.authentihash`.

File: rfcs/text/0014-extend-file-pe.md [hunk 10]
```diff
@@ -77,17 +78,217 @@ Stage 1: Provide a high-level description of example sources of data. This does
...
+    },
+    "rich_pe_header": {
```
Comment (ebeahan) [inline rfcs/text/0014-extend-file-pe.md:178]: `rich_pe_header` is another new one. What's defined right now in the RFC is `rich_header.hash.md5`

**Review by ebeahan — COMMENTED** [review]
I focused most of my review on the first mapping example. There's numerous fields that don't align with what's been proposed so far.

File: rfcs/text/0014-extend-file-pe.md (hunk 2 above)
Comment (peasead) [reply rfcs/text/0014-extend-file-pe.md:81]: Unintentionally. Removed.

File: rfcs/text/0014-extend-file-pe.md (hunk 3 above)
Comment (peasead) [reply rfcs/text/0014-extend-file-pe.md:87]: @ebeahan do you mean that in `0014-extend-file-pe.md` there's this example
```
"file": {
  "pe": {
    "machine_type": 332,
    "flattened": {
      "debug": [
        {
...
```
and in `pe.yml` it's reflected like this
```
---
- name: pe

  fields:

    - name: icon.hash.dhash
      level: extended
      type: keyword
      description: >
        Difference Hash (dhash) to find files with a visually similar icon or thumbnail.

      example: b806e17c8e330d82

    - name: debug
      level: extended
      type: nested
      description: >
        Debug information, if present
...
```

@dcode, was the intention to be `file.pe.{debug,resources}.*` or `file.pe.flattened.{debug,resources}.*`

File: rfcs/text/0014-extend-file-pe.md (hunk 4 above)
Comment (peasead) [reply rfcs/text/0014-extend-file-pe.md:123]: @dcode were the comments added from your dataset or is that part of the PE file?

File: rfcs/text/0014-extend-file-pe.md [hunk 11]
```diff
@@ -85,9 +86,10 @@ Stage 3: Add more real world example source documents so we have at least 2 tota
 
 ## Scope of impact
 
-There should be no breaking changes, depreciation strategies, or significant refactoring as this is extending the existing fieldset.
+There should be no breaking changes, depreciation strategies, or significant refactoring as this is creating a sub-field for the existing file. fieldset.
```
Comment (devonakerr) [inline rfcs/text/0014-extend-file-pe.md:89]: For consistency of language, consider replacing "the existing file. fieldset" with "the `file` top-level fieldset"

**Review by devonakerr — APPROVED** [review]
I made one suggestion that the author should consider optional, and which pertains to consistent language.

File: rfcs/text/0014-extend-file-pe.md (hunk 11 above)
Comment (peasead) [inline rfcs/text/0014-extend-file-pe.md:89]:
```suggestion
There should be no breaking changes, depreciation strategies, or significant refactoring as this is creating a sub-field for the `file` top-level fieldset.
```

File: rfcs/text/0014-extend-file-pe.md (hunk 3 above)
Comment (peasead) [reply rfcs/text/0014-extend-file-pe.md]: Questions caused be stale data examples. Refactoring.

File: rfcs/text/0014-extend-file-pe.md (hunk 4 above)
Comment (peasead) [reply rfcs/text/0014-extend-file-pe.md]: Questions caused by stale data examples. Refactoring.

File: rfcs/text/0014-extend-file-pe.md (hunk 5 above)
Comment (peasead) [reply rfcs/text/0014-extend-file-pe.md]: Questions caused by stale data examples. Refactoring.

File: rfcs/text/0014-extend-file-pe.md (hunk 6 above)
Comment (peasead) [reply rfcs/text/0014-extend-file-pe.md]: Questions caused by stale data examples. Refactoring.

File: rfcs/text/0014-extend-file-pe.md (hunk 7 above)
Comment (peasead) [reply rfcs/text/0014-extend-file-pe.md]: Questions caused by stale data examples. Refactoring.

File: rfcs/text/0014-extend-file-pe.md (hunk 8 above)
Comment (peasead) [reply rfcs/text/0014-extend-file-pe.md]: Questions caused by stale data examples. Refactoring.

File: rfcs/text/0014-extend-file-pe.md (hunk 9 above)
Comment (peasead) [reply rfcs/text/0014-extend-file-pe.md]: Questions caused by stale data examples. Refactoring.

File: rfcs/text/0014-extend-file-pe.md (hunk 10 above)
Comment (peasead) [reply rfcs/text/0014-extend-file-pe.md]: Questions caused by stale data examples. Refactoring.

**Comment by peasead** [comment]
Closing until ready to revisit this.

**Pull request closed without merging**